src/Framework/AspNetCoreAnalyzers/test/Http/HeaderDictionaryAddFixerTests.cs

```diff
@@ -0,0 +1,276 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.CodeAnalysis.Testing;
+using VerifyCS = Microsoft.AspNetCore.Analyzers.Verifiers.CSharpCodeFixVerifier<
+    Microsoft.AspNetCore.Analyzers.Http.HeaderDictionaryAddAnalyzer,
+    Microsoft.AspNetCore.Analyzers.Http.Fixers.HeaderDictionaryAddFixer>;
+
+namespace Microsoft.AspNetCore.Analyzers.Http;
+
+public class HeaderDictionaryAddFixerTests
+{
+    private const string AppendCodeActionEquivalenceKey = "Use IHeaderDictionary.Append";
+    private const string IndexerCodeActionEquivalenceKey = "Use the indexer";
+
+    [Fact]
+    public async Task IHeaderDictionary_WithAdd_FixedWithAppend()
+    {
+        // Arrange & Act & Assert
+        await VerifyCS.VerifyCodeFixAsync(@"
+using Microsoft.AspNetCore.Http;
+namespace HeaderDictionaryAddFixerTests;
+public class Program
+{
+    public static void Main()
+    {
+        var context = new DefaultHttpContext();
+        {|#0:context.Request.Headers.Add(""Accept"", ""text/html"")|};
+    }
+}",
+        new[]
+        {
+            new DiagnosticResult(DiagnosticDescriptors.DoNotUseIHeaderDictionaryAdd)
+                .WithLocation(0)
+                .WithMessage(Resources.Analyzer_HeaderDictionaryAdd_Message)
+        },
+        @"
+using Microsoft.AspNetCore.Http;
+namespace HeaderDictionaryAddFixerTests;
+public class Program
+{
+    public static void Main()
+    {
+        var context = new DefaultHttpContext();
+        context.Request.Headers.Append(""Accept"", ""text/html"");
+    }
+}",
+        codeActionEquivalenceKey: AppendCodeActionEquivalenceKey);
+    }
+
+    public static IEnumerable<object[]> FixedWithAppendAddsUsingDirectiveTestData()
+    {
+        yield return new[]
+        {
+            @"
+using Microsoft.AspNetCore.Mvc;
+
+namespace HeaderDictionaryAddFixerTests;
+public class TestController : ControllerBase
+{
+    public IActionResult Endpoint()
+    {
+        {|#0:Response.Headers.Add(""Content-Type"", ""text/html"")|};
```

Comment:
Consider testing this case:

```suggestion
        {|#0:Response.Headers.Add(""Content-Type"", ""text/html"")|};
        {|#1:Response.Headers.Add(""Content-Type"", ""text/html"")|};
```